post_request_task/task.py

```diff
@@ -13,11 +13,13 @@
 
 
 _locals = threading.local()
+_locals.task_queue = []
+_locals.queue_tasks = False
 
 
 def _get_task_queue():
     """Returns the calling thread's task queue."""
-    return _locals.__dict__.setdefault('task_queue', [])
+    return _locals.task_queue
```

Comment:
I still kinda like it for external apps that want to look at the queue without knowing how it's stored behind the scenes. Only our task module should know how it's implemented.